### PR TITLE
Fix #365: preserve __proto__ keys via null-prototype parsed objects

### LINKED ISSUE #365: Minify and Format silently drop __proto__ keys via nodeToValue plain-object recipient
## Summary

`src/app/core/json/parse.ts:174-182` uses a plain `{}` recipient in `nodeToValue`. When the parser encounters a `__proto__` key in user input, the key is assigned to `obj.__proto__` which invokes the prototype setter on `Object.prototype` -- the key is silently dropped from the returned value rather than retained as an own enumerable key.

This affects every consumer that reads the parsed value back out, including:

- **Minify** (`HomeComponent.onMinify` -> re-serializes via `JSON.stringify` on the parsed value)
- **Format** (`HomeComponent.onFormat` -> same path)
- **Toolbar Sort** (new in #361 - uses the same parser path)

The right-click Sort path (also new in #361) uses a byte-splicing patcher that sidesteps `nodeToValue` entirely, so it is unaffected.

## Repro

1. Paste `{"__proto__": 1, "a": 2}` into the editor.
2. Click Format (or Minify, or the toolbar Sort).
3. Observe: the `__proto__` key is gone from the output.

## Fix

Change the `nodeToValue` recipient at `parse.ts:174-182` (or the equivalent line range at fix time) from `{}` to `Object.create(null) as Record<string, unknown>`. Add a spec covering `__proto__` round-trip through Minify and Format.

## Priority

`priority:low` - `__proto__` is unusual in user JSON; the symptom is "key disappears" rather than a security issue (the prototype itself is never mutated, only set on the local object, and JavaScript's prototype-setter semantics block prototype pollution from plain assignment in modern engines). But the silent drop is surprising and easy to fix.

Filed from #361 plan rubber-duck.

### PR DESCRIPTION
**Pull request by geevensingh**
## Summary

Fixes #365. `nodeToValue` allocated a plain `{}` recipient, so an `obj['__proto__']` assignment invoked `Object.prototype`'s `__proto__` setter rather than creating an own property. The key was silently dropped from `parseResult().value`, which made:

- **Minify** drop the key from its `JSON.stringify` output, and
- the **JsonTree** miss the row entirely (`Object.keys` returned nothing).

`Format` was never affected (text-based via `jsoncFormat` + `applyEdits`); the new spec is a regression guard.

## Fix

Introduce an exported `createJsonObject()` helper in `parse.ts` returning `Object.create(null)`, and use it as the `nodeToValue` recipient. The helper's JSDoc documents the invariant for any future code that reconstructs parsed-JSON-shaped objects (e.g. #361's eventual sort path), so the bug cannot silently re-appear at the next call site.

## Tests

10 new specs across 4 files:

- `parse.spec.ts` (6 tests): own-key retention, nested-object, nested-array-element, `__proto__: null` edge case, `JSON.stringify` round-trip, **Jasmine deep-equality pin** (regression guard against future Jasmine versions tightening the constructor-mismatch gate -- the H1 finding from the rubber-duck review).
- `build-tree.spec.ts` (1 test): tree renders the `__proto__` child row.
- `home.component.spec.ts` (2 tests): `onMinify` preserves the key; `onFormat` regression guard.
- `title-suggester.service.spec.ts` (1 test): strategies handle null-prototype parsed inputs end-to-end (the existing `inputFor` uses `JSON.parse`, which returns plain objects, so the existing coverage didn't exercise the production path after the fix).

## Verification (Definition of Done)

- `npm run lint:all`: PASS (lockfile drift caught and synced on first run).
- `npm test`: **2981 / 2982 SUCCESS** (1 pre-existing skip; H1 Jasmine-equality concern empirically refuted -- every existing `toEqual(parseResult().value)` keeps passing).
- `npm --prefix api test`: **465 / 465 SUCCESS**.
- `npm run build`: PASS (1.01 MB initial bundle).
- **Perf gate** (`perf:l1` before/after on the same machine): **zero regressions exceeding 5%**. `parse.deep25.10k` -46.0% and `parse.mixed-d10.380k` -25.3% (faster, likely V8 dictionary-mode allocation tradeoff or first-run noise). All build-tree deltas within +/-5%. No fallback to Option B needed.

## SemVer

Patch bump `1.0.0` -> `1.0.1`. Bug fix restoring intended behavior on an edge-case input; no API or stored-shape change.

## Alternatives considered

- **Option B**: `Object.defineProperty` for `__proto__` only -- conditionally rejected. Kept as the fallback if perf measurement showed regression; perf cleared, so unused.
- **Option C**: filter `__proto__` keys at parse time -- rejected. That IS the current broken behavior.

## Closes

Closes #365.

---
**Session**
- AI-Local: `ecea1723-9daa-4144-9ffa-6ed6445df083`
- AI-Cloud: `24f96145-9ead-44d9-9a60-b051a50d79f9`

### CODE REVIEW

File: src/app/core/json/parse.spec.ts [hunk 1]
```diff
@@ -51,6 +51,68 @@ describe('parse (pure)', () => {
...
+      // `parseResult().value` returns null-prototype objects after
+      // the #365 fix; existing specs at home.component.spec.ts:1020
+      // and :1147 use `expect(...).toEqual({ ... })` with plain-object
+      // literal expectations. Jasmine 6.2's `eq_` short-circuits the
+      // constructor-mismatch gate when either constructor is
+      // `undefined` (which is the case for null-prototype objects).
+      // Mirror that exact pattern -- a null-prototype LHS deep-equal
+      // to a plain object literal RHS -- so a future Jasmine upgrade
+      // that tightens the gate fails this spec first instead of
+      // breaking the broader suite silently.
```
[the remark this answers is not in the document]
Comment (geevensingh) [reply src/app/core/json/parse.spec.ts]: Good catch -- and they were already stale within this same PR (the new `__proto__` tests above shifted the callsites from :1020/:1147 to :1040/:1167). Replaced with the stable test names "onValueChange does not update treePaneInputs synchronously when editor is non-empty" and "onFormat preserves the existing parseResult shape so tree selection can survive" plus a generic search-string fallback in commit 6bf841a.
